atflow/network.py

```diff
@@ -95,52 +102,60 @@ def get_convolution_op(input_shape, output_shape, kernel_shape):
             with tf.name_scope(name):
-                if padding_type=='VALID' and np.sum(padding) > 0:
-                    inputs = tf.pad(inputs, padding, name='padding')
-                return tf.nn.conv2d(inputs, weight, strides, padding_type, name='convolution')
+                if padding == 'VALID' and np.sum(pad_vals) > 0:
+                    inputs = image_pad(inputs, pad_vals, mode=padding_type, name='padding')
+                return conv2d(inputs, weight, strides,
+                              padding=padding,
+                              padding_type=padding_type,
+                              name='convolution')
+
 
     else:
         def conv_op(inputs, weight, name='generic_convolution'):
-            if padding_type=='SAME':
+            if padding == 'SAME':
                 padded_output = [padded_shape[0]] + output_shape[-3:]
             else:
                 padded_output = padded_shape
             with tf.name_scope(name):
                 if padded_output[0] is None:
                     batch_size = tf.shape(inputs)[0]
                     padded_output = [batch_size] + padded_output[1:]
-
-                output = tf.nn.conv2d_transpose(inputs, weight, padded_output, strides, padding_type, name='transpose_convolution')
-                if padding_type=='VALID' and np.sum(padding) > 0:
-                    output = tf.slice(output, [0, padding[1][0], padding[2][0], 0],
+                output = tf.nn.conv2d_transpose(inputs, weight, padded_output, strides, padding, name='transpose_convolution')
+                if padding == 'VALID' and np.sum(pad_vals) > 0:
+                    output = tf.slice(output, [0, pad_vals[1][0], pad_vals[2][0], 0],
                                       [-1] + output_shape[-3:], name='cropping')
+                if not output.shape.is_fully_defined():
+                    output.set_shape([None] + output_shape[-3:])
                 return output
 
     return filter_shape, conv_op
 
 
 
 
-def normalize_weights(w, dims=(0,), bias=1e-5):
+def normalize_weights(w, dims=(0,), bias=1e-12):
     """
     L2 normalize weights of the given tensor along specified dimension(s).
 
     Args:
         w: Tensor to be normalized
         dims: dimension(s) along which to normalize the Tensor. Defaults to (0,)
-        bias: Bias value added to the computed norm to prevent dividing by 0. Defaults to 1e-5
+        bias: Bias value added to the computed norm to prevent dividing by 0. Defaults to 1e-12
 
     Returns: Tensor of same type and shape as `w` whose norm is set to approximately 1 along the specificed dimensions
     """
     with tf.name_scope('normalization'):
         return w / (tf.sqrt(tf.reduce_sum(tf.square(w), dims, keep_dims=True) + bias))
 
 
-def weight_variable(shape, name='weight', mean=0.0, stddev=None, initializer=None, constrain=None, dtype=tf.float32):
+def weight_variable(shape, name='weight', mean=0.0, stddev=1e-3, initializer=None, constrain=None, dtype=tf.float32):
```

Comment:
The last commit to the master branch changed this to raise an error if stddev not provided, did you mean to revert it or did that change happened after you forked this branch